From: Jesse Sung <jesse.sung@canonical.com>
Subject: Shutdown the IO channel for the HFP server on exit.

When the rfkill switch is toggled, the channel is unref'd before being closed,
which lead to issues with the interfaces properly getting brought back up
when the rfkill switch gets back to a non-blocked state.

--- audio/manager.c	2012-03-21 16:27:23.000000000 +0800
+++ audio/manager.c	2012-03-21 16:29:52.340242119 +0800
@@ -990,6 +990,7 @@
 	}
 
 	if (adp->hfp_hs_server) {
+		g_io_channel_shutdown(adp->hfp_hs_server, TRUE, NULL);
 		g_io_channel_unref(adp->hfp_hs_server);
 		adp->hfp_hs_server = NULL;
 	}
